examples/run_smc_on_biphenyl.py

```diff
@@ -0,0 +1,59 @@
+import argparse
+from datetime import datetime
+from pickle import dump
+
+from timemachine.fe.absolute_hydration import set_up_ahfe_system_for_smc
+from timemachine.md.smc import sequential_monte_carlo
+from timemachine.testsystems.ligands import get_biphenyl
+
+
+def parse_options():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--n_walkers", type=int, help="number of walkers", default=100)
+    parser.add_argument("--n_windows", type=int, help="number of lambda windows", default=100)
+    parser.add_argument("--n_md_steps", type=int, help="number of MD steps per move", default=100)
+    parser.add_argument("--resample_thresh", type=float, help="resample when fractional ESS < thresh", default=0.6)
+    parser.add_argument("--debug_mode", type=bool, help="save full trajectories", default=False)
+
+    cmd_args = parser.parse_args()
+
+    if cmd_args.debug_mode:
+        n_frames = cmd_args.n_walkers * cmd_args.n_windows
+        print(f"Warning! This will take a lot of disk space! ({n_frames} simulation frames)")
+
+    return cmd_args
+
+
+def save_smc_result(smc_result, save_full_trajectories=False):
+    uid = f"{datetime.now()}"
```

Comment:
nit: if this is to be used in a filename, a format without spaces might be easier to deal with, e.g.
```python
datetime.now().isoformat(timespec="seconds")
```
or
```python
datetime.now().strftime("%Y%m%dT%H%M%S"))
```

Reply:
This seems almost identical to `save_smc_result` in `run_smc_on_freesolv.py`. Could this take `uid` as an argument for consistency, and would it make sense to then merge?

Reply:
Applied in https://github.com/proteneer/timemachine/pull/718#discussion_r854309159

Reply:
Another thought -- not necessary for this PR, but it seems like it might be nice for reproducibility to compute the output UID from the timemachine version and a hash of the inputs, rather than a timestamp. Would something like this make sense here?